plugin/storage/kafka/writer.go

```diff
@@ -57,8 +57,10 @@ func NewSpanWriter(
 	}()
 	go func() {
 		for e := range producer.Errors() {
-			logger.Error(e.Err.Error())
-			writeMetrics.SpansWrittenFailure.Inc(1)
+			if e != nil && e.Err != nil {
+				logger.Error(e.Err.Error())
+				writeMetrics.SpansWrittenFailure.Inc(1)
```

Comment:
This line should not be within the `if`.

Reply:
Done